devel/libenkf/src/hook_workflow.c

```diff
@@ -0,0 +1,224 @@
+/*
+   Copyright (C) 2012  Statoil ASA, Norway.
+
+   The file 'hook_workflow.c' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <ert/util/util.h>
+#include <ert/util/subst_list.h>
+
+#include <ert/config/config_parser.h>
+
+#include <ert/job_queue/workflow.h>
+
+#include <ert/enkf/config_keys.h>
+#include <ert/enkf/hook_workflow.h>
+#include <ert/enkf/ert_workflow_list.h>
+#include <ert/enkf/runpath_list.h>
+
+#define RUN_MODE_PRE_SIMULATION_NAME     "PRE_SIMULATION"
+#define RUN_MODE_POST_SIMULATION_NAME    "POST_SIMULATION"
+
+#define HOOK_WORKFLOW_NAME  "HOOK WORKFLOW"
+#define QC_WORKFLOW_NAME    "QC WORKFLOW"
+#define RUNPATH_LIST_FILE   ".ert_runpath_list"
+
+
+struct hook_workflow_struct {
+  hook_run_mode_enum       run_mode;
+  char                   * path;
+  workflow_type          * workflow;
+  ert_workflow_list_type * workflow_list;
+  runpath_list_type      * runpath_list;
+};
+
+
+hook_workflow_type * hook_workflow_alloc( ert_workflow_list_type * workflow_list , const char * path ) {
+  hook_workflow_type * hook_workflow = util_malloc( sizeof * hook_workflow );
+  hook_workflow->run_mode = PRE_SIMULATION;
```

Comment:
The idea here is that a manager is a collection of hooks and that each hook should know when it should be run.